parlai/core/logs.py

```diff
@@ -94,3 +96,102 @@ def log_metrics(self, setting, step, report):
 
     def flush(self):
         self.writer.flush()
+
+
+class WandbLogger(object):
+    """
+    Log objects to Weights and Biases.
+    """
+
+    @classmethod
+    def add_cmdline_args(
+        cls, parser: ParlaiParser, partial_opt: Optional[Opt] = None
+    ) -> ParlaiParser:
+        """
+        Add w&b CLI args.
+        """
+        logger = parser.add_argument_group('Tensorboard Arguments')
+        logger.add_argument(
+            '-wblog',
+            '--wandb-log',
+            type='bool',
+            default=False,
+            help="Enable W&B logging of metrics, default is %(default)s",
+            hidden=False,
+        )
+        logger.add_argument(
+            '--wandb-name', type=str, default=None, help='W&B run name', hidden=True
+        )
+
+        logger.add_argument(
+            '--wandb-project',
+            type=str,
+            default=None,
+            help='W&B project name. Defaults to timestamp.',
+            hidden=False,
+        )
+        return logger
+
+    def __init__(self, opt: Opt, model=None):
+        try:
+            # tensorboard is a very expensive thing to import. Wait until the
+            # last second to import it.
+            import wandb
+
+        except ImportError:
+            raise ImportError('Please run `pip install wandb`.')
+
+        name = opt.get('wandb_name')
+        project = opt.get('wandb_project') or datetime.datetime.now().strftime(
```

Comment:
1) where is this `wandb_project` ever be set in opt? 
2) Also another problem here, if it is set by timestamps, what if my job gets pre-empted, would it setup a new `wandblogger`  then in this case?

Reply:
1. In our sweep setup, I have this parameter automatically set to the sweep name. So we should be good
2. Good question. Let's try it and see what happens.